src/core/Settings.js

```diff
@@ -916,8 +916,28 @@ import Events from './events/Events.js';
  * The version of the CMCD to use.
  *
  * If not specified this value defaults to 1.
+ * @property {Array.<CmcdTarget>} [targets]
+ * List of CMCD reporting targets.
  */
 
+/**
+ * @typedef {Object} CmcdTarget
+ * @property {string} [mode]
+ * Mode of the CMCD reporting.
+ * @property {boolean} [enabled]
+ * Whether the CMCD reporting is enabled for this target.
+ * @property {string} [url]
+ * The reporting endpoint URL.
+ * @property {string} [events]
+ * The events that should trigger the CMCD reporting.
+ * @property {string} [timeInterval]
+ * The time interval for the CMCD reporting in event mode. The 't' event should be setted in the events array to use this parameter.
+ * @property {Array.<string>} [enabledKeys]
+ * CMCD keys to include in the report.
+ * @property {Array.<string>} [includeOnRequests]
+ * Types of requests CMCD should be included on (e.g., 'mpd', 'segment'). 
```

Comment:
Instead of DASH specific terms like `mpd`, should this just be a list of [`CmcdObjectType`](https://github.com/streaming-video-technology-alliance/common-media-library/blob/main/lib/src/cta/CmObjectType.ts) values?